File: flutter/lib/queries.dart
import 'package:todolist/TemplateString.dart';

const String createTableQuery = """
  CREATE TABLE IF NOT EXISTS ToDoList 
  (
    id INTEGER PRIMARY KEY AUTOINCREMENT, 
    description TEXT,
    date TEXT,
    isDone INTEGER
  ); """;

const String selectRecordsQuery =
    'SELECT id, description, date, isDone FROM ToDoList';

const String deleteRecordsQuery = 'DELETE FROM ToDoList';

final deleteRecordQuery = TemplateString('DELETE FROM ToDoList WHERE id={id}');

final insertNewTaskQuery = TemplateString("""
  INSERT INTO ToDoList (description, date, isDone) 
  VALUES ('{toDoTextString}', '{timeString}', {isDoneINT})""");

final updateTaskQuery = TemplateString("""
  UPDATE ToDoList SET 
    description = '{toDoTextString}',
    date = '{timeString}',
    isDone = {isDoneINT}
  WHERE id={taskID}""");

File: react_native/AlarmProject/lib/queries.js
const createTableQuery =
  'CREATE TABLE IF NOT EXISTS alarms (id INTEGER PRIMARY KEY, status INT, notificationIN INT, radio INT);';

const insertQuery =
  'INSERT INTO alarms (id, status, notificationIN, radio) VALUES (?, ?, ?, ? )';

const selectQuery =
  'SELECT id, status, notificationIN, radio FROM alarms ORDER BY notificationIN';

const deleteQuery = 'DELETE FROM alarms WHERE id = ?';

const selectLastIDQuery = 'SELECT max(id) AS ID FROM alarms';

const updateAlarmQuery =
  'Update alarms SET (status, notificationIN, radio) = (?, ?, ?) WHERE id = ?';

const updateStatusQuery = 'Update alarms SET (status) = (?) WHERE id = ?';

export {
  createTableQuery,
  insertQuery,
  selectQuery,
  deleteQuery,
  selectLastIDQuery,
  updateAlarmQuery,
  updateStatusQuery,
};
